ValveResourceFormat/Resource/ResourceTypes/KV3.cs

```diff
@@ -0,0 +1,592 @@
+                stateStack.Pop();
+                objStack.Peek().AddProperty(currentName, new KVValue(typeof(String), currentString.ToString()));
+                return;
+            }
+
+            currentString.Append(c);
+        }
+
+        //Reading multiline string
+        private static void ReadValueStringMulti(char c)
+        {
+            //Check for ending
+            if (c == '"' && c == '"' && fileStream[index + 1] == '"' && fileStream[index + 2] == '"' && fileStream[index-1] != '\\')
+            {
+                stateStack.Pop();
+                objStack.Peek().AddProperty(currentName, new KVValue(typeof(String), currentString.ToString()));
+
+                //Skip to end
+                index += 2;
+                return;
+            }
+
+            currentString.Append(c);
+        }
+
+        //Read a boolean variable
+        private static void ReadValueBool(char c)
+        {
+            //Stop reading once the end of true or false is reached
+            if (c == 'e')
+            {
+                currentString.Append(c);
+                stateStack.Pop();
+                objStack.Peek().AddProperty(currentName, new KVValue(typeof(Boolean), currentString.ToString() == "true" ? true : false));
+                return;
+            }
+
+            currentString.Append(c);
+        }
+
+        //Read a numerical value
+        private static void ReadValueNumber(char c)
+        {
+            //For arrays
+            if (c == ',')
+            {
+                stateStack.Pop();
+                stateStack.Push(State.SEEK_VALUE);
+                if ( currentString.ToString().Contains('.')) {
+                    objStack.Peek().AddProperty(currentName, new KVValue(typeof(Double), Double.Parse(currentString.ToString(), CultureInfo.InvariantCulture)));
+                } else {
+                    objStack.Peek().AddProperty(currentName, new KVValue(typeof(Int32), Int32.Parse(currentString.ToString(), CultureInfo.InvariantCulture)));
+                }
+                return;
+            }
+
+            //Stop reading the number once whtiespace is encountered
+            if (whiteSpace.Contains(c))
+            {
+                stateStack.Pop();
+                //Distinguish between doubles and ints
+                if (currentString.ToString().Contains('.'))
+                {
+                    objStack.Peek().AddProperty(currentName, new KVValue(typeof(Double), Double.Parse(currentString.ToString(), CultureInfo.InvariantCulture)));
+                }
+                else
+                {
+                    objStack.Peek().AddProperty(currentName, new KVValue(typeof(Int32), Int32.Parse(currentString.ToString(), CultureInfo.InvariantCulture)));
+                }
+                return;
+            }
+
+            currentString.Append(c);
+        }
+
+        //Read an array
+        private static void ReadValueArray(char c)
+        {
+            //This shouldn't happen
```

Comment:
If this shouldn't happen, why is this a `Console.WriteLine` instead of an exception? And why use `Console.WriteLine`?